web/src/app/forms/FormField.tsx

```diff
@@ -1,14 +1,124 @@
-import React, { useContext, useEffect } from 'react'
-import p from 'prop-types'
+import React, { ElementType, useContext, useEffect, ReactNode } from 'react'
```

Comment:
Move ts conversions and refactors to their own PRs

This one, in particular, needs a LOT of careful consideration -- I'd recommend we build something new to replace how we're doing forms instead of trying to refactor it (every time we've tried, we've found countless regressions)

Reply:
This file is basically mandatory for this PR to pass. I can do all this work separated out I guess.

Reply:
If we want to do that, we won't be able to update material for awhile, since that will take some time.